### PR TITLE
Sprint 7 cascade delete

### PR DESCRIPTION
**Pull request by purplesmurf1998**
# Description
Simple cascade delete feature for the Vehicle model. Deletes the Comments, History and VehicleSale objects that are related to the vehicle being deleted.
# How to test

1. Delete a vehicle

# Jira User Story
[[AP-111] - When a vehicle is deleted, cascade on delete (delete all the objects associated with it)](https://soen390-team12.atlassian.net/browse/AP-111?atlOrigin=eyJpIjoiOGU5ZTczZGFjOGRmNGQwNjk0OWM3NjlkNDcyYzBjMmYiLCJwIjoiaiJ9)

### CODE REVIEW

File: backend/models/Vehicle.js [hunk 1]
```diff
@@ -35,4 +38,11 @@ const VehicleSchema = new mongoose.Schema({
   properties: Object,
 })
 
+// cascade delete comments, history, vehiclesale
+VehicleSchema.pre('remove', async function(next) {
+  await Comment.deleteMany({ vehicle: this._id });
+  await History.deleteMany({ vehicle: this._id });
+  await VehicleSale.deleteMany({ vehicle: this._id });
```
Comment (AbdulSir) [inline backend/models/Vehicle.js:45]: Looks good but I think we're missing to also delete all the events that are associated with the vehicles.

**Review by AbdulSir — CHANGES_REQUESTED** [review]
When I deleted a vehicle, the events that were associated with the deleted vehicles remained in the database.

File: backend/models/Vehicle.js (hunk 1 above)
Comment (purplesmurf1998) [reply backend/models/Vehicle.js:45]: Right that's a good point, will add this

Comment (purplesmurf1998) [reply backend/models/Vehicle.js:46]: Added in this [commit](https://github.com/purplesmurf1998/Autotracks/pull/86/commits/cafd5ae8a3a1782cea668e1bd3f3b2a898b8ffcc)

**Review by AbdulSir — APPROVED** [review]
This looks good for merging

**Review by AnthonyChraim — APPROVED** [review]
Working as intended

**Pull request merged**